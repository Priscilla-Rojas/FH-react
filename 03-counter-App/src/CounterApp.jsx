import { useState } from 'react';
import PropTypes from 'prop-types';
// import { getImagen } from './base--pruebas/11-async-await';


export const CounterApp = ({value}) => {

  const [ counter, setCounter] = useState( value );
  // console.log(getImagen)

  const handleAdd = () => setCounter( counter + 1 )
  const handleSustract = () => setCounter( counter - 1 )
  const handleReset = () => setCounter( value )


  return (
    <>
      <h1> CounterApp </h1>
      <h2> { counter } </h2>
      <button onClick={ handleAdd }> +1 </button>
      <button onClick={ handleSustract }> -1 </button>
      <button aria-label='btn-reset' onClick={ handleReset }> Reset </button>
    </>
  )
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired
}