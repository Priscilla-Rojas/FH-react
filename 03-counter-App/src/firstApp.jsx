import React from 'react';
import PropTypes from 'prop-types';

const FirstApp = ( {name, title, subtitle} ) => {

  return (
    <>
      <h1 data-testid='test-title'>{ title }</h1>
      <p>{ subtitle }</p>
      <p>{ name }</p>
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired
}
FirstApp.defaultProps = {
  name: 'Priscilla Rojas',
  title: 'no hay titulo',
  subtitle: 'no hay subtitulo'
}
export default FirstApp