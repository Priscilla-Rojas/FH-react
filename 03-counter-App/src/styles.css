html, body{
  background-color: #21232a;
  color: white;
  font-family: Helvetica, Arial, sans-serif;
}


button {
  padding: 5px 10px;
  font-size: 1rem;
  border-radius: 5px;
  background-color: pink;
  border: none;
  margin-right: 5px;
}