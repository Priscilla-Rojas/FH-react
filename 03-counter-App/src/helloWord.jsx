import {CounterApp} from './CounterApp';

export const App = ()=>{
  return (
    <>
      {/* <h1>Hello, this is App componment</h1> */}
    </>
  );
}
