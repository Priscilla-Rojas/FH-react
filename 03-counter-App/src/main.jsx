import React from 'react'
import ReactDom from 'react-dom/client';
import './styles.css'
import { CounterApp } from './CounterApp';
import FirstApp from './firstApp';
FirstApp

ReactDom.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <CounterApp value={ 0 }></CounterApp>
    {/* <FirstApp></FirstApp> */}
  </React.StrictMode>
)