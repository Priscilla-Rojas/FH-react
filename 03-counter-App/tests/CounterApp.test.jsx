import { fireEvent, render, screen } from "@testing-library/react"
import { CounterApp } from "../src/CounterApp";

describe('Pruebas en <CounterApp/>', () => {
  const valueInicial = 0;
  test('Debe Hacer match con el Snapshot', () => { 

    const { container } = render( < CounterApp value={ valueInicial }/> )
    expect ( container ).toMatchSnapshot();
  })

  test('debe mostar el valor inicial de 100', ()=>{

    render( < CounterApp value={ 100 }/> )
    // expect( screen.getByText( '100')).toBeTruthy();
    expect( screen.getByRole( 'heading',{ level: 2 } ).innerHTML).toContain('100');
  })

  test('debe incrementar con el boton +1', () => { 
    
    render( <CounterApp value={valueInicial}/>);
    fireEvent.click(screen.getByText('+1'));
    expect(screen.getByText('1')).toBeTruthy();
  })

  test('debe decrementar con el boton -1', () => { 
    
    render( <CounterApp value={valueInicial}/>);
    fireEvent.click(screen.getByText('-1'));
    expect(screen.getByText('-1')).toBeTruthy();
  })
})