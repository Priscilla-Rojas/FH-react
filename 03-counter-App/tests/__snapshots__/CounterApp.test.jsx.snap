// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp/> Debe Hacer match con el Snapshot 1`] = `
<div>
  <h1>
     CounterApp 
  </h1>
  <h2>
     
    0
     
  </h2>
  <button>
     +1 
  </button>
  <button>
     -1 
  </button>
  <button>
     Reset 
  </button>
</div>
`;
